tests/test_micropub.php

```diff
@@ -256,6 +256,43 @@ function syndicate_to() {
 		$this->assertNull( static::$after_micropub_args );
 	}
 
+
+	function test_custom_query() {
+		function custom_query( $return, $input ) {
+			if ( 'abc' === $input['q'] ) {
+				return array( 'abc' => array( '123' ) );
+			}
+			return $return;
+		}
+		add_filter( 'micropub_query', 'custom_query', 10, 2 );
+		$_GET['q'] = 'abc';
+		$expected = array( 'abc' => array( '123' ) );
+		$this->check( 200, $expected );
+
+		$this->assertEquals( $_GET, static::$before_micropub_input );
+		$this->assertEquals( $_GET, static::$after_micropub_input );
+		$this->assertNull( static::$after_micropub_args );
+	}
+
+
+	function test_custom_query_still_returns_default() {
+		function custom_default_query( $return, $input ) {
```

Comment:
this should be a class method and resused in both tests instead of duplicated.

alternatively, feel free to just move lines 286-288 into the previous test, and remove this one.